src/error.js

```diff
@@ -319,14 +319,19 @@ function reportErrorToServer(message, filename, line, col, error) {
   const data = getErrorReportData(message, filename, line, col, error,
       hasNonAmpJs);
   if (data) {
-    // Report the error to viewer if it has the capability. The data passed
-    // to the viewer is exactly the same as the data passed to the server
-    // below.
-    maybeReportErrorToViewer(this, data);
     reportingBackoff(() => {
-      const xhr = new XMLHttpRequest();
-      xhr.open('POST', urls.errorReporting, true);
-      xhr.send(JSON.stringify(data));
+      // Report the error to viewer if it has the capability. The data passed
+      // to the viewer is exactly the same as the data passed to the server
+      // below.
+      maybeReportErrorToViewer(this, data).then(reportedErrorToViewer => {
+        if (!reportedErrorToViewer) {
+          // Only report to StackDriver if it's not reported to viewer, as
+          // such XHRs will triger CSP violation for AMP4EMAIL.
```

Comment:
nit. triger -> trigger.

Reply:
Done. 

All tests are passing after this change. Apparently, there is no test covering the XHR firing part. 

I was trying to add some but didn't find a good example :/

Reply:
+1. Tests would be nice given that we are introducing multiple code paths.

Reply:
Done added some unit tests.
Also involves some refactor to make testing easier (or possible). PTAL